src/connection-state-handler.ts

```diff
@@ -113,6 +122,32 @@ export class ConnectionStateHandler extends EventEmitter<ConnectionStateEventHan
    * @returns Current connection state.
    */
   public getConnectionState(): ConnectionState {
-    return this.mediaConnectionState;
+    const { connectionState } = this.getCurrentStatesCallback();
+
+    const state = connectionState[0].toUpperCase() + connectionState.slice(1);
+
+    return state as ConnectionState;
+  }
+
+  /**
+   * Gets current ice connection state.
+   *
+   * @returns Current ice connection state.
+   */
+  public getIceConnectionState(): IceConnectionState {
+    const { iceState } = this.getCurrentStatesCallback();
+
+    const state = iceState[0].toUpperCase() + iceState.slice(1);
+
+    return state as IceConnectionState;
```

Comment:
I'm not sure why we need to define our own `IceConnectionState` enum. Wouldn't it be better to have `RTCIceConnectionState` as the return type for `getIceConnectionState()` and `RTCPeerConnectionState` as return type for `getConnectionState()`? Then we wouldn't have to be doing these tricks with converting first letter to uppercase etc. which I'm not a huge fan of....

Reply:
Done